x-pack/plugins/cases/public/components/all_cases/columns.tsx

```diff
@@ -52,23 +54,95 @@ const renderStringField = (field: string, dataTestSubj: string) =>
   field != null ? <span data-test-subj={dataTestSubj}>{field}</span> : getEmptyTagValue();
 
 interface GetCasesColumn {
-  actions: Array<DefaultItemIconButtonAction<Case>>;
   caseDetailsNavigation: CasesNavigation<CaseDetailsHrefSchema, 'configurable'>;
   filterStatus: string;
-  isModal: boolean;
+  handleIsLoading: (a: boolean) => void;
+  refreshCases?: (a?: boolean) => void;
+  showActions: boolean;
+  showCaseTitleAsHref?: boolean;
 }
-export const getCasesColumns = ({
-  actions,
+export const useCasesColumns = ({
   caseDetailsNavigation,
   filterStatus,
-  isModal,
+  handleIsLoading,
+  refreshCases,
+  showActions,
+  showCaseTitleAsHref = true,
 }: GetCasesColumn): CasesColumns[] => {
-  const columns = [
+  const { loading: isLoadingCases, dispatchUpdateCaseProperty } = useGetCases();
```

Comment:
move actions for deleting/updating a single case into columns, along with the confirmDeleteModal

Reply:
dope